src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -191,145 +191,71 @@ public ICompilationTracker FreezePartialState(CancellationToken cancellationToke
             {
                 var state = this.ReadState();
 
-                // If we're already finalized then just return what we have, and with the frozen bit flipped so that
-                // any future forks keep things frozen.
-                if (state is FinalCompilationTrackerState finalState)
-                {
-                    // If we're finalized and already frozen, we can just use ourselves.
-                    return finalState.IsFrozen
-                        ? this
-                        : new CompilationTracker(this.ProjectState, finalState.WithIsFrozen(), this.SkeletonReferenceCache.Clone());
-                }
-
-                Contract.ThrowIfFalse(state is null or InProgressState);
-
-                GetPartialCompilationState(
-                    state,
-                    out var inProgressProject,
-                    out var compilationWithoutGeneratedDocuments,
-                    out var compilationWithGeneratedDocuments,
-                    out var generatorInfo,
-                    cancellationToken);
-
-                // Transition us to a frozen in progress state.  With the best compilations up to this point, but no
-                // more pending actions, and with the frozen bit flipped so that any future forks keep things
-                // frozen.
-                var inProgressState = InProgressState.Create(
-                    isFrozen: true,
-                    compilationWithoutGeneratedDocuments,
-                    generatorInfo,
-                    compilationWithGeneratedDocuments,
-                    ImmutableList<(ProjectState, CompilationAndGeneratorDriverTranslationAction)>.Empty);
-
-                return new CompilationTracker(inProgressProject, inProgressState, this.SkeletonReferenceCache.Clone());
-            }
-
-            public ICompilationTracker FreezePartialStateWithDocument(
-                DocumentState docState,
-                CancellationToken cancellationToken)
```

Comment:
no more freezing partial state with documents.  instead, we just freeze the whole state, and then add/update the doc of interest back at the outer level